config/printer-creatbot-f430-v9.5.cfg

```diff
@@ -0,0 +1,245 @@
+homing_speed: 100
+homing_retract_dist: 5.0
+
+[stepper_z]
+step_pin: PF2
+dir_pin: PF3
+enable_pin: !PK0
+microsteps: 16
+rotation_distance: 4
+endstop_pin: probe:z_virtual_endstop
+position_max: 300
+##physical endstops if required##
+#endstop_pin: ^PC4 
+#endstop_pin: ^PD2 
+#position_endstop: 0.5
+
+## Extruders  ##
+ 
+[extruder]
+step_pin: PA0
+dir_pin: !PA1
+enable_pin: !PA2
+microsteps: 16
+rotation_distance: 22.5
+full_steps_per_rotation: 200
+nozzle_diameter: 0.400
+filament_diameter: 1.750
+heater_pin: PE4
+sensor_type: AD595
+sensor_pin: PK7
+#control: pid
+#pid_Kp: 12.9
+#pid_Ki: 0.76
+#pid_Kd: 55
+min_temp: 0
+max_temp: 265
+
+[gcode_macro T0]
+gcode:
+    SET_GCODE_OFFSET X=-36
+    ACTIVATE_EXTRUDER EXTRUDER=extruder
+
+[extruder1]
+step_pin: PA3
+dir_pin: !PA4
+enable_pin: !PA5
+microsteps: 16
+rotation_distance: 22.5
+full_steps_per_rotation: 200
+nozzle_diameter: 0.400
+filament_diameter: 1.750
+heater_pin: PE5
+sensor_type: AD595
+sensor_pin: PK6
+#control: pid
+#pid_Kp: 12.9
+#pid_Ki: 0.76
+#pid_Kd: 55
+min_temp: 0
+max_temp: 400
+
+[gcode_macro T1]
+gcode:
+    SET_GCODE_OFFSET X=36
+    ACTIVATE_EXTRUDER EXTRUDER=extruder1
+
+## Heaters ##
+
+[heater_bed]
+heater_pin: PH6
+sensor_type: EPCOS 100K B57560G104F
+sensor_pin: PK4
+#control: pid
+#pid_Kp: 345.78
+#pid_Ki: 14.59
+#pid_Kd: 2407.99
+min_temp: 0
+max_temp: 130
+
+[heater_generic chamber_heater]
+gcode_id: C
+heater_pin: PH3
+max_power: 1
+sensor_type: EPCOS 100K B57560G104F
+sensor_pin: PK3
+#smooth_time:
+control: pid
+pid_Kp: 54.027
+pid_Ki: 0.770
+pid_Kd: 948.182
+#pwm_cycle_time:
+min_temp: 0
+max_temp: 75
+
+[verify_heater chamber_heater]
+max_error: 120
+check_gain_time: 120
+hysteresis: 5
+heating_gain: 1
+
+[gcode_macro M138] ##Chamber Heater ON
+gcode:
+    SET_HEATER_TEMPERATURE HEATER=chamber_heater TARGET={ params.T }
+
+## Fans ##
+
+[fan]
+# Print cooling fan
+pin: PH4
+kick_start_time: 0.500
+
+[heater_fan hotend_fan]
+# Hotend fan
+pin: PH5
+kick_start_time: 0.500
+heater: extruder
+heater_temp: 50.0
+
+[heater_fan exhaust_fan]
+# Exhaust fan
+pin: PB4
+kick_start_time: 0.500
+heater: heater_bed
+heater_temp: 60.0
+
+## Filament sensors ##
+
+[filament_switch_sensor e0_sensor]
+pause_on_runout: True
+switch_pin: PC1
+
+[filament_switch_sensor e1_sensor]
+pause_on_runout: True
+switch_pin: PC3
+
+## Output Pins ##
+
+[output_pin caselight]
+pin: PB7
+pwm: False
+value: 1
+
+[output_pin power]
+pin: !PG1
+value: 1
+shutdown_value: 0
+
+[output_pin beeper]
+pin: PL5
+
+## Probe / Levelling ##
+
+[bltouch]
+x_offset: 0
+y_offset: 72
+sensor_pin: ^PC4
+control_pin: PE3
+speed: 30
+z_offset: 1.2
+pin_up_touch_mode_reports_triggered: False
+
+[safe_z_home]
+home_xy_position: 200,100 # Change coordinates to the center of your print bed
+speed: 100
+z_hop: 10                 # Move up 10mm
+z_hop_speed: 5
+
+[bed_mesh]
+speed: 50
+horizontal_move_z: 5
+mesh_min: 20,75
+mesh_max: 380, 275
+probe_count: 5,5
+algorithm: bicubic
+relative_reference_index: 12

```

Comment:
Is this needed? Typically not set when using a probe as Z endstop as it overrides the mesh z-offset

Reply:
Z endstop only probes the same point and if the bed is not flat and need a mesh?

Reply:
See https://www.klipper3d.org/Bed_Mesh.html?#the-relative-reference-index

Typically only relevant if you have a physical endstop for the general homing and a probe for bed mesh. If you use the probe for both functionalities it is not needed. Creating and using a bed mesh works anyway.

Reply:
sorry do you just mean line 233 and not the whole bed mesh section?

Reply:
Only relative_reference_index

Reply:
ok i will remove it. any idea why the white space check is failing?

Reply:
Looks like windows line endings instead of unix